docs/dev/cut-a-new-release.md

```diff
@@ -1,64 +1,230 @@
-# Glean Release Process
+# Glean development & release process
 
-These are the steps needed to cut a new release from latest master.
+Glean consists of multiple libraries for different platforms and targets.
+The main supported libraries are releases as one.
+The following document describes the development and release process.
+Development happens on the main repository <https://github.com/mozilla/glean>.
 
-1. Update the changelog.
+The development & release process roughly follows the [GitFlow model](https://nvie.com/posts/a-successful-git-branching-model/).
+
+> **Note:** The rest of this section assumes that `upstream` points to the `https://github.com/mozilla/glean` repository,
+> while `origin` points to the developer fork.
+> For some developer workflows, `upstream` can be the same as `origin`.
+
+**Table of Contents**:
+
+* [Development](#development)
+* [Regular release](#regular-release)
+* [Hotfix release for latest version](#hotfix-release-for-latest-version)
+* [Hotfix release for previous version](#hotfix-release-for-previous-version)
+
+## Published artifacts
+
+* The Kotlin libraries are published: [GitHub Releases](https://github.com/mozilla/glean/releases), [Mozilla Maven](https://maven.mozilla.org/?prefix=maven2/org/mozilla/telemetry/).
+* Python bindings are published on PyPI: [glean-sdk](https://pypi.org/project/glean-sdk/).
+* iOS framework artifacts: [GitHub Releases](https://github.com/mozilla/glean/releases).
+* Rust crates are published on crates.io: [glean-core](https://crates.io/crates/glean-core), [glean-ffi](https://crates.io/crates/glean-ffi).
+
+## Development
```

Comment:
This section is very related to (but doesn't really overlap with) the [Sending Pull Requests](https://mozilla.github.io/glean/book/contributing.html#sending-pull-requests) docs.  Maybe we should merge the information here and then just link from there to here?

Reply:
What about the other way around? Keep Development/Sending Pull Request separate over there and link to the release process?

Reply:
That seems fine, too :)